File: solutions/go/JumpGame.go
func canJump(nums []int) bool {
	maxReach := 0

	for i := 0; i <= maxReach; i++ {
		if tmp := nums[i] + i; tmp > maxReach {
			maxReach = tmp
		}

		if maxReach >= len(nums)-1 {
			return true
		}
	}

	return false
}

File: solutions/cpp/JumpGame.cc
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReach = 0;

        for (int i = 0; i <= maxReach; ++i) {
            maxReach = std::max(maxReach, i+nums[i]);
            if (maxReach >= nums.size()-1) {
                return true;
            }
        }

        return false;
    }
};
